packages/smooth_app/lib/pages/folksonomy/folksonomy_card.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:flutter/material.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/helpers/product_cards_helper.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_page.dart';
+import 'package:smooth_app/pages/folksonomy/folksonomy_provider.dart';
+import 'package:smooth_app/pages/folksonomy/tag.dart';
+import 'package:smooth_app/themes/constant_icons.dart';
+
+class FolksonomyCard extends StatelessWidget {
+  const FolksonomyCard(this.product);
+  final Product product;
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider(
+      create: (_) => FolksonomyProvider(product.barcode!),
+      child: Container(child: Card(product)),
+    );
+  }
+}
+
+class Card extends StatelessWidget {
+  const Card(this.product);
+  final Product product;
+
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+      onTap: () {
+        Navigator.of(context)
+            .push(
+              MaterialPageRoute(
+                builder: (BuildContext context) => FolksonomyPage(product),
+              ),
+            )
+            .then((completion) => {
+                  Provider.of<FolksonomyProvider>(context, listen: false)
+                      .fetchProductTags(),
+                });
+      },
+      child: Padding(
+        padding: const EdgeInsets.only(top: LARGE_SPACE),
+        child: buildProductSmoothCard(
+          title: Stack(
+            children: <Widget>[
+              Center(child: Text('Product Tags')),
+            ],
+          ),
+          body: Container(
+            width: double.infinity,
+            padding: const EdgeInsetsDirectional.all(LARGE_SPACE),
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisAlignment: MainAxisAlignment.start,
+              children: <Widget>[
+                CardList(),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class CardList extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final provider = context.watch<FolksonomyProvider>();
+
+    if (provider.isLoading) {
+      return const Center(child: CircularProgressIndicator.adaptive());
+    } else if (!provider.tagsExist) {
+      return Center(
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const Expanded(
+              child: Text(
+                'No tags found. Tags can be used to better group similar products. Tap to add.', // TODO: localize
```

Comment:
@teolemon Not sure on the verbase you would want here when there are not tags on a product.